models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+      iaf_cond_alpha_mc_kinetics::PROX > );
+  insert_( Name( "g_ex.p" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::G_EXC,
+      iaf_cond_alpha_mc_kinetics::PROX > );
+  insert_( Name( "g_in.p" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::G_INH,
+      iaf_cond_alpha_mc_kinetics::PROX > );
+  insert_( Name( "i_ap.p" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::I_AP,
+      iaf_cond_alpha_mc_kinetics::PROX > );
+
+  insert_( Name( "V_m.d" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::V_M,
+      iaf_cond_alpha_mc_kinetics::DIST > );
+  insert_( Name( "g_ex.d" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::G_EXC,
+      iaf_cond_alpha_mc_kinetics::DIST > );
+  insert_( Name( "g_in.d" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::G_INH,
+      iaf_cond_alpha_mc_kinetics::DIST > );
+  insert_( Name( "i_ap.d" ),
+    &iaf_cond_alpha_mc_kinetics::get_y_elem_< iaf_cond_alpha_mc_kinetics::State_::I_AP,
+      iaf_cond_alpha_mc_kinetics::DIST > );
+
+  insert_( names::t_ref_remaining, &iaf_cond_alpha_mc_kinetics::get_r_ );
+  insert_( names::threshold, &iaf_cond_alpha_mc_kinetics::get_th_ );
+
+  // changes
+  insert_( names::ca_spike_count, &iaf_cond_alpha_mc_kinetics::get_ca_ );
+  insert_( names::ca_current, &iaf_cond_alpha_mc_kinetics::get_ica_ );
+}
+}
+
+/* ----------------------------------------------------------------
+ * Iteration function
+ * ---------------------------------------------------------------- */
+
+extern "C" int
+nest::iaf_cond_alpha_mc_kinetics_dynamics( double, const double y[], double f[], void* pnode )
+{
+  // some shorthands
+  typedef nest::iaf_cond_alpha_mc_kinetics N;
+  typedef nest::iaf_cond_alpha_mc_kinetics::State_ S;
+
+  // get access to node so we can work almost as in a member function
+  assert( pnode );
+  const nest::iaf_cond_alpha_mc_kinetics& node =
+    *( reinterpret_cast< nest::iaf_cond_alpha_mc_kinetics* >( pnode ) );
+
+  // compute dynamics for each compartment
+  // computations written quite explicitly for clarity, assume compile
+  // will optimized most stuff away ...
+  for ( size_t n = 0; n < N::NCOMP; ++n )
+  {
+    // membrane potential for current compartment
+    const double V = y[ S::idx( n, S::V_M ) ];
+
+    // excitatory synaptic current
+    const double I_syn_exc = ( y[ S::idx( n, S::G_EXC ) ] ) * ( V - node.P_.E_ex[ n ] );
+    // inhibitory synaptic current
+    const double I_syn_inh = ( y[ S::idx( n, S::G_INH ) ] ) * ( V - node.P_.E_in[ n ] );
+
+    // leak current
+    const double I_L = node.P_.g_L[ n ] * ( V - node.P_.E_L[ n ] );
+
+    // changes
+
+    double minf = 0.0;
+    double hinf = 0.0;
+    if ( n == N::DIST )
+    {
+      if ( node.P_.act_flag == 1.0 )
```

Comment:
Make flag bool, then this becomes `if ( node.P_.act_flag )`, or even better with improved variable names `if ( node.P_.ca_spikes_active )`